### DISCUSSION TITLE
Auto imports in external plugins

### DISCUSSION THREAD

**Discussion by guzz** [nuxt/nuxt#16728] [Questions]
I just finished building a module (https://github.com/guzz/nuxt-simple-i18n), and it is all working and all, but one of the issues I had when publishing to npm (https://www.npmjs.com/package/nuxt-simple-i18n) was to understand that the auto imports do not work in external plugins.

I don't know if this is by design but I think this information should be somewhere in the documentation.

Also, I still think I'm doing something wrong because my plugin imports from `#app` is being linted in vstudio as non existent althought it works when you install it in a nuxt app.

![image](https://user-images.githubusercontent.com/1380036/174485206-aca793c1-1d99-4463-9d7c-5bb1065d2b65.png)

**Comment by Flanker72** [comment]
I had the similar issue with my GraphQL plugin. However, I solved it by exporting `provide` from the plugin. However, it seems, you defined your plugin the right way.

Please, check that all the required steps are completed: https://v3.nuxtjs.org/guide/going-further/modules - especially, `npm run dev:prepare`.

Hope, somebody will help to resolve this issue, seems some is missing in the documentation. I haven't enough knowledge to investigate a problem now. You may try to have a look on other modules, compatible with Nuxt3.

**Comment by dan-hammond** [comment]
I have noticed this issue as well. Everything seems to work in the playground project created by the template, but once it's packed and installed in another website project, none of the auto imports within the module work. 

As a workaround I've had to go through every file in my module and add explicit imports.

Edit: I messaged the Nuxt Discord about this issue and was advised by Daniel Roe that explicit imports are necessary within modules.
